torch/csrc/jit/graph_executor.cpp

```diff
@@ -193,7 +193,7 @@ struct DifferentiableGraphOp {
         // Note: we have to set this up in place, or we have to throw away and
         // reallocate variables that were already created in wrapTensors. We
         // should add an API for this.
-        Variable output = outputs[idx].toTensor();
+        Variable output = toOptionalTensor(outputs[idx]);
```

Comment:
This should have a huge warning around it. We use Undefined tensor in the autograd in a way that it is never used inside python and allowing a None to turn into an undefined tensor here should be well documented. I'd prefer it actually be explicit:
```
Variable output;
if (!outputs[idx].isNone()) {
   output = outputs[idx].toTensor();
}
```